proposals/reward-proposal-v2-reward-distribution.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Epoch Reward Distribution
+---
+
+## Problem
+
+In the `Epoch Reward Calculation V2`, we proposed a new method to handle
+calculating epoch rewards. This is the second part of that proposal - `Reward
+Distribution`.
+
+## Proposed Solutions
+
+Similar to [Partitioned Inflationary Rewards
+Distribution](https://github.com/solana-labs/solana/blob/master/docs/src/proposals/partitioned-inflationary-rewards-distribution.md),
+the rewards are distributed over `M` slots.
+
+To address the concern for the lack of double-entry booking for reward
+distribution, a special system instruction, `DistributeReward`, will be
+introduced to distribute the rewards from the family of `EpochRewardReserve`
+accounts and the `EpochRewardProof` system account.
+
+This instruction takes the target reward account, the amount to distribute, the
+`EpochRewardReserve` account, and the merkle proof for the distribution. When
+executing this instruction, it first verifies the supplied merkle proof against
+the proof stored in in `EpochRewardProof`. If the verification passed, it will
+transfer the reward from the `EpochRewardReserve` account to the target reward
+account.
```

Comment:
Again, this whole merkle part is pretty ambiguous and incomplete.
* If a node wants to distribute rewards for a particular stake account where does it obtain the necessary proof?
   * This is actually related to the previous question that where the leaves of the merkle tree are stored? 
* What stops someone from reusing the same merkle proof to double deposit rewards?
   * What exactly happens once the rewards is successfully distributed for the 1st time that stops a 2nd double distribution?